modules/core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/api/blobholder/TestBlobHolderAdapterService.java

```diff
@@ -45,25 +45,28 @@
 @Deploy("org.nuxeo.ecm.core.api")
 public class TestBlobHolderAdapterService {
 
-    @Inject
-    protected BlobHolderAdapterService service;
-
     @Inject
     protected HotDeployer hotDeployer;
 
+    protected BlobHolderAdapterService getService() {
+        return Framework.getService(BlobHolderAdapterService.class);
+    }
+
     @Test
     public void testService() {
-        assertTrue(service instanceof BlobHolderAdapterComponent);
+        assertTrue(getService() instanceof BlobHolderAdapterComponent);
     }
 
     @Test
     public void testContrib() throws Exception {
-        BlobHolderAdapterComponent bhac = (BlobHolderAdapterComponent) service;
+        BlobHolderAdapterComponent bhac = (BlobHolderAdapterComponent) getService();
 
         assertEquals(0, bhac.getFactoryNames().size());
 
         hotDeployer.deploy(Constants.CORE_TEST_TESTS_BUNDLE + ":test-blob-holder-adapters-contrib.xml");
 
+        // retrieve service again after hotreload
+        bhac = (BlobHolderAdapterComponent) getService();
```

Comment:
Or you could keep all the injection code and just do `service = Framework.getService(BlobHolderAdapterService.class)` here (keeping the comment)